Client/Frontend/Browser/Playlist/Cache/PlaylistDownloadManager.swift

```diff
@@ -0,0 +1,524 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AVFoundation
+import Shared
+import Data
+
+private let log = Logger.browserLogger
+
+protocol PlaylistDownloadManagerDelegate: AnyObject {
```

Comment:
1. I don't see a limit on the cache download? Should we add a limit say 1gb, to avoid the app from freezing up trying to cache a large video?

2. Will this cache be cleared with if the user clears private data using `Clear Private Data` > `Cache`

Reply:
The app won't freeze on a large video because we download it in a background thread via Apple's APIs. However, I guess we can add a size limit, but not sure what that size should be as we can even download 4K videos and 1080P videos which can easily hit 1GB.